### PR TITLE
Now displays ingredients, instructions properly

### PR DESCRIPTION
**Pull request by ddsnijders**
Displays ingredients and instructions from the database

### CODE REVIEW

File: classes/RecipeRequester.php [hunk 1]
```diff
@@ -9,13 +9,17 @@ public function __construct(){
...
 
-    public function requestRecipeByID(string $id): Recipe{
+    public function requestRecipeByID(int $id): Recipe{
         return $this->databaseManager->getRecipe($id);
     }
 
     public function requestAllRecipes(): array{
         return $this->databaseManager->getAllRecipes();
     }
+
```
Comment (JNBetteraij) [inline classes/RecipeRequester.php]: since you're not using this function, go ahead and delete it

File: database/databaseManager.php [hunk 2]
```diff
@@ -245,7 +246,16 @@ public function getAllIngredientsFromRecipe($id): array{
         private function createRecipeObject(array $data): Recipe{
             $values = [];
             foreach($data as $key => $row){
-                    array_push($values, $row);
```
Comment (JNBetteraij) [inline database/databaseManager.php:248]: would like to rewrite this one together with you

File: database/databaseManager.php [hunk 3]
```diff
@@ -340,7 +350,7 @@ public function deleteEntryInTable(string $tableName, int $id): void{
     
     //echo var_export($dbm->deleteRecipe(1), true);
 
-    // $dbm = new DatabaseManager();
+    $dbm = new DatabaseManager();
```
Comment (JNBetteraij) [inline database/databaseManager.php]: re-comment this

File: public/displayComponents/RecipeDisplayer.php [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+            $this->recipes = $this->getAllRecipes();
+        }
+
+        public function displayRecipeList(){
+            $recipes = $this->recipes;
+            $convertedValues = "";
+            foreach ($recipes as $recipe){
+                $convertedValues .= $this->convertRecipeToThumbnail($recipe);
+            }
+            return $convertedValues;
```
Comment (JNBetteraij) [inline public/displayComponents/RecipeDisplayer.php]: remove this function from this class

File: public/displayComponents/RecipeDisplayer.php [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+        }
+
+        private function convertRecipeToThumbnail(Recipe $recipe){
+
+            $replace = ["{id}"];
+            $value = [$recipe->getID()];
+
+            $template = file_get_contents("displayComponents/recipeThumbnail.html");
+            return str_replace($replace, $value, $template);
+        }
```
Comment (JNBetteraij) [inline public/displayComponents/RecipeDisplayer.php]: remove from this class

File: public/displayComponents/RecipeDisplayer.php [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+        
+        public function getThumbnailElement(int $id): string{
+        
+            $replace = ["{id}"];
+            $values = ["$id"];
+        
+            $template = file_get_contents("displayComponents/recipeThumbnail.html");
+            return str_replace($replace, $values, $template);
+        }
```
Comment (JNBetteraij) [inline public/displayComponents/RecipeDisplayer.php]: remove this function

**Review by JNBetteraij — CHANGES_REQUESTED** [review]
clean up where commented, the rest looks good

File: database/databaseManager.php (hunk 2 above)
Comment (ddsnijders) [reply database/databaseManager.php:248]: Sure, it might be a little hacky

**Pull request closed without merging**